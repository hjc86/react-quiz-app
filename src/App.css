* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  font-family: 'Exo', sans-serif;
  font-weight: 100px;
}

.container {
  display: flex;
  flex-direction: column;
  
}