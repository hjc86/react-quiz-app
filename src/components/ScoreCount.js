import React from 'react';
import styles from './ScoreCount.module.css';

class ScoreCount extends React.Component {
    constructor(props) {
        super(props)
    }


    render() {
        return (
            <div className={styles.scoreCountList}>
                 {this.props.scoreCount}
            </div>
                
        )
    }
}

export default ScoreCount;