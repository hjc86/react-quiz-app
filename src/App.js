import React from 'react';
import UserInfo from './components/UserInfo';
import Quiz from './components/Quiz';
import Score from './components/Score';
import './App.css';


class App extends React.Component {

// states GetUsername, GetAPI, GetQuestions, GetAnswers, GetScore
constructor(props){
  super(props);
  this.state = {
      QuizQ : [],
      Answers : [],
      userName: null,
      difficulty: null,
      category: null
  };
}

// Fetch API
getQuizApi = async (userData) => {

  const categoryNumber = userData.category;
  const difficulty = userData.difficulty;
  const apiCall = await fetch(`https://opentdb.com/api.php?amount=5&category=${categoryNumber}&difficulty=${difficulty}&type=multiple`);
  const data = await apiCall.json();

  console.log(userData);
 
  // to get the whole data 
  this.setState({
    QuizQ : data.results
  
  });
  console.log(data);
}


getUserData =(userDataObj) =>{
  this.setState(
    userDataObj
  ) 
  // console.log("before update",this.state)
}

componentDidUpdate(){
  // console.log("after update",this.state)
}


formSubmitted = (state) => {
  this.setState({
      userName: state.username,
      difficulty: state.difficulty,
      category: state.category
  })
  console.log(state);
  this.getQuizApi(state);
  
}



    render() {
      return (
        <div className="container">
          <UserInfo userData={this.getUserData} onSubmit={this.formSubmitted}/>
          {this.state.QuizQ.map(questions => {
            const answers = [questions.correct_answer, ...questions.incorrect_answers]
            console.log(answers);
            return <Quiz questions={questions.question} answers={questions.correct_answer} incorrect={questions.incorrect_answers}/>
          })
          }
          <Score />
        </div>
      );
}

}

export default App;

// answers={questions.correct_answer}