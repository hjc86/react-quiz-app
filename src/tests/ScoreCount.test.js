import React, { Component } from 'react';
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme';
import App from '../App';
import renderer from 'react-test-renderer';
import ScoreCount from '../components/ScoreCount';



describe("<ScoreCount/>", () => {
    
    it('mounts to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<ScoreCount />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    let wrapper;
    beforeEach(() => wrapper = shallow(<ScoreCount />));
    
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render a <div />', () => {
        expect(wrapper.find('div').length).toEqual(1);
    });

    

})