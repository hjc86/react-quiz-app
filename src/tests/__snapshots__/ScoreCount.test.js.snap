// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreCount/> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<ScoreCount/> renders correctly 1`] = `
<div
  className="scoreCountList"
/>
`;
